File: android/app/src/main/java/vn/team/freechat/request/GetContactsRequest.java
package vn.team.freechat.request;

import com.tvd12.ezyfoxserver.client.entity.EzyData;
import com.tvd12.ezyfoxserver.client.factory.EzyEntityFactory;
import com.tvd12.ezyfoxserver.client.request.EzyRequest;

import vn.team.freechat.constant.Commands;

/**
 * Created by Dung Ta Van on 11/3/18.
 * Copyright © 2018 Young Monkeys. All rights reserved.
 **/

public class GetContactsRequest implements EzyRequest {

    private final int skip;
    private final int limit;

    public GetContactsRequest(int skip, int limit) {
        this.skip = skip;
        this.limit = limit;
    }

    @Override
    public EzyData serialize() {
        return EzyEntityFactory.newObjectBuilder()
                .append("skip", skip)
                .append("limit", limit)
                .build();
    }

    @Override
    public Object getCommand() {
        return Commands.CHAT_GET_CONTACTS;
    }
}

File: android/app/src/main/java/com/tvd12/freechat/request/GetContactsRequest.kt
package com.tvd12.freechat.request

import com.tvd12.ezyfoxserver.client.entity.EzyData
import com.tvd12.ezyfoxserver.client.factory.EzyEntityFactory
import com.tvd12.ezyfoxserver.client.request.EzyRequest
import com.tvd12.freechat.constant.Commands

/**
 * Created by tavandung12 on 10/3/18.
 */

class GetContactsRequest(
    private val skip: Int,
    private val limit: Int
) : EzyRequest {

    override fun serialize() : EzyData {
        return EzyEntityFactory.newObjectBuilder()
                .append("skip", skip)
                .append("limit", limit)
                .build()
    }

    override fun getCommand() : String {
        return Commands.CHAT_GET_CONTACTS
    }
}
